### ISSUE TITLE
API error

### ISSUE THREAD

**Issue by Captainkyle420** [hjdhjd/homebridge-myq#1119]
**Describe The Problem:**

It wouldn’t connect to myQ server. Has worked perfectly fine up till  10/9/23. User and pass is correctly configured. Look online at forums and it seems like they are blaming an api change from myQ.

**To Reproduce:**

**Logs:**

[12/10/2023, 07:02:301
[Garage Door MyQ]
djusting myl API normal refresh duration rom 302 to 300. Setting too high of a norm al refresh duration is strongly discourage d due to myl occasionally blocking account s who overtax the myl API.
[12/10/2023, 07:02:301 [Garage Door Myl] D ebug logging on. Expect a lot of data. [12/10/2023, 07:02:30] Homebridge v1.6.1 ( HAP v0.11.1) (Garage Door MyQ) is running on port 44864.
[12/10/2023, 07:02:301 [Garage Door Myl] m
YO API: Temporary
myl API server-side issu
es encountered: 429 - Too Many Requests. R etrying the API call.
[12/10/2023, 07:02:301 [Garage Door Myl] m
YO API: Switching
to myl cloud region: eas
t.
[12/10/2023, 07:02:301 [Garage Door Myl] m
YO API: Temporary
myl API server-side issu
es encountered: 429 -
Too Many Requests. R
etrying the API call.
[12/10/2023, 07:02:30] [Garage Door Myl] m
Ye API: Switching to myl cloud region: wes
t.
[12/10/2023, 07:02:301 [Garage Door Mye] m
Ye API: Temporary myl
API server-side issu
es encountered: 429 -
Too Many Requests. R
etrying the API call.
[12/10/2023, 07:02:301 [Garage Door Myl] m
YO API: Switching to myl cloud region: aut
[12/10/2023, 07:02:301 [Garage Door Myl] m
YO API
error: Temporary myl API server-sid
e issues encountered: 429 - Too Many Reque stS .
[12/10/2023, 07:02:301 [Garage Door Myl] m
YO
API: Unable to access the Auth authori zation endpoint.
[12/10/2023, 07:02:301 [Garage Door Myl] m
YO
API: throttling
refreshDevices API call
• Using cached data from the past two seco nds.

**Homebridge Configuration:**

{
    "bridge": {
        "name": "Homebridge 88A5",
        "username": "redacted",
        "port": 51646,
        "pin": "redacted",
        "advertiser": "avahi"
    },
    "accessories": [
        {
            "accessory": "Roku",
            "name": "Roku",
            "ip": "redacted",
            "inputs": [
                {
                    "id": "31012",
                    "name": "Vudu Movie & TV Store"
                },
                {
                    "id": "tvinput.hdmi1",
                    "name": "Xbox"
                },
                {
                    "id": "tvinput.hdmi2",
                    "name": "PlayStation"
                },
                {
                    "id": "tvinput.hdmi3",
                    "name": "HDMI 3"
                },
                {
                    "id": "tvinput.hdmi4",
                    "name": "HDMI 4 (ARC)"
                },
                {
                    "id": "12",
                    "name": "Netflix"
                },
                {
                    "id": "61322",
                    "name": "HBO Max"
                },
                {
                    "id": "20454",
                    "name": "FOX NOW"
                },
                {
                    "id": "13842",
                    "name": "Vudu"
                },
                {
                    "id": "837",
                    "name": "YouTube"
                },
                {
                    "id": "2285",
                    "name": "Hulu"
                },
                {
                    "id": "73386",
                    "name": "MTV"
                },
                {
                    "id": "2213",
                    "name": "Roku Media Player"
                },
                {
                    "id": "28",
                    "name": "Pandora"
                },
                {
                    "id": "8838",
                    "name": "SHOWTIME"
                },
                {
                    "id": "35059",
                    "name": "HISTORY"
                },
                {
                    "id": "556230",
                    "name": "Night Landscapes"
                }
            ],
            "info": {
                "udn": "redacted",
                "serialNumber": "redacted",
                "deviceId": "redacted",
                "advertisingId": "redacted",
                "vendorName": "Insignia",
                "modelName": "Insignia NS-55DR710NA17",
                "modelNumber": "6303X",
                "modelRegion": "US",
                "isTv": "true",
                "isStick": "false",
                "screenSize": "55",
                "panelId": "3",
                "mobileHasLiveTv": "true",
                "uiResolution": "1080p",
                "tunerType": "ATSC",
                "supportsEthernet": "true",
                "wifiMac": "redacted",
                "wifiDriver": "bcm",
                "hasWifiExtender": "false",
                "hasWifi5GSupport": "true",
                "canUseWifiExtender": "true",
                "ethernetMac": "redacted",
                "networkType": "wifi",
                "networkName": "Lord of the Pings",
                "friendlyDeviceName": "INSIGNIA•Roku TV -",
                "friendlyModelName": "INSIGNIA•Roku TV",
                "defaultDeviceName": "INSIGNIA•Roku TV - ",
                "userDeviceName": "Roku TV",
                "userDeviceLocation": "",
                "buildNumber": "15C.50E04312A",
                "softwareVersion": "11.5.0",
                "softwareBuild": "4312",
                "secureDevice": "true",
                "language": "en",
                "country": "US",
                "locale": "en_US",
                "timeZoneAuto": "true",
                "timeZone": "US/Mountain",
                "timeZoneName": "United States/Mountain",
                "timeZoneTz": "America/Denver",
                "timeZoneOffset": "-420",
                "clockFormat": "24-hour",
                "uptime": "248797",
                "powerMode": "PowerOn",
                "supportsSuspend": "true",
                "supportsFindRemote": "false",
                "supportsAudioGuide": "true",
                "supportsRva": "true",
                "hasHandsFreeVoiceRemote": "false",
                "developerEnabled": "false",
                "keyedDeveloperId": "",
                "searchEnabled": "true",
                "searchChannelsEnabled": "true",
                "voiceSearchEnabled": "true",
                "notificationsEnabled": "false",
                "notificationsFirstUse": "true",
                "supportsPrivateListening": "true",
                "supportsPrivateListeningDtv": "true",
                "supportsWarmStandby": "true",
                "headphonesConnected": "false",
                "supportsAudioSettings": "false",
                "expertPqEnabled": "1.0",
                "supportsEcsTextedit": "true",
                "supportsEcsMicrophone": "true",
                "supportsWakeOnWlan": "true",
                "supportsAirplay": "false",
                "hasPlayOnRoku": "true",
                "hasMobileScreensaver": "false",
                "supportUrl": "www.insigniaproducts.com",
                "grandcentralVersion": "9.0.90",
                "davinciVersion": "2.8.20",
                "avSyncCalibrationEnabled": "3.0"
            },
            "_bridge": {
                "username": "redacted",
                "port": 41251
            }
        },
        {
            "name": "Dummy Garage Door Close",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 60000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 56120
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Garage Door Open",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 60000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 31101
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Disarm",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 60000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 46506
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Arm Home",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 60000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 48846
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy18650",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 21600000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 43450
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "DummyDoorBell",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 900000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 38609
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Bathroom Fan 1hr",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 3600000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 41422
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Arm Away",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 60000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted”,
                "port": 42508
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Porch 15min",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 900000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 59401
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Bathroom Mirror 5min",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 300000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 43792
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Bedroom Ceiling 5min",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 300000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 41333
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Kitchen UC 5min",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 300000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 57945
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy 22000",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 21600000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 42191
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Foyer Stairs 5min",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 300000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 57632
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Suspend Motion Detection",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": false,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 37376
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "Dummy Loft Stairs 5min",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 300000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": "redacted",
                "port": 59350
            }
        },
        {
            "name": "Dummy BHRM Motion Suspended",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": "redacted",
                "port": 54163
            }
        },
        {
            "name": "Dummy KTN Motion Suspended",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": "redacted",
                "port": 49372
            }
        },
        {
            "name": "Dummy FYRSW Motion Suspended",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": "redacted",
                "port": 56261
            }
        },
        {
            "name": "Dummy LFTSW Motion Suspended",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": "redacted",
                "port": 56814
            }
        },
        {
            "name": "Dummy BDRMMotion Suspended",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": "redacted",
                "port": 49192
            }
        },
        {
            "name": "Dummy LFT Motion Suspended",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": “redacted ",
                "port": 56861
            }
        },
        {
            "name": "Dummy PRH Motion Suspended",
            "stateful": true,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "accessory": "DummySwitch",
            "_bridge": {
                "username": "redacted",
                "port": 48007
            }
        },
        {
            "name": "Temp Maintenance ",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 1000,
            "random": false,
            "resettable": false,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 30109
            },
            "accessory": "DummySwitch"
        },
        {
            "name": "H2OFountainAutoRestart",
            "stateful": false,
            "dimmer": false,
            "reverse": false,
            "time": 10000,
            "random": false,
            "resettable": true,
            "brightness": 0,
            "disableLogging": false,
            "_bridge": {
                "username": "redacted",
                "port": 59014
            },
            "accessory": "DummySwitch"
        }
    ],
    "platforms": [
        {
            "name": "Config",
            "port": 8581,
            "platform": "config"
        },
        {
            "mqtt": {
                "base_topic": "zigbee2mqtt",
                "server": "mqtt://redacted",
                "reject_unauthorized": false,
                "keepalive": 60,
                "version": 4,
                "disable_qos": false
            },
            "defaults": {
                "exclude": false
            },
            "exclude_grouped_devices": false,
            "devices": [
                {
                    "id": "redacted",
                    "converters": {
                        "switch": {
                            "type": "switch"
                        },
                        "occupancy": {
                            "type": "motion"
                        }
                    }
                },
                {
                    "id": "redacted",
                    "converters": {
                        "switch": {
                            "type": "switch"
                        },
                        "occupancy": {
                            "type": "motion"
                        }
                    }
                },
                {
                    "id": "redacted",
                    "converters": {
                        "switch": {
                            "type": "switch"
                        },
                        "occupancy": {
                            "type": "motion"
                        }
                    }
                },
                {
                    "id": "redacted",
                    "converters": {
                        "switch": {
                            "type": "switch"
                        },
                        "occupancy": {
                            "type": "motion"
                        }
                    }
                },
                {
                    "id": "redacted",
                    "converters": {
                        "switch": {
                            "type": "switch"
                        },
                        "occupancy": {
                            "type": "motion"
                        }
                    }
                },
                {
                    "id": "redacted",
                    "converters": {
                        "switch": {
                            "type": "switch"
                        },
                        "occupancy": {
                            "type": "motion"
                        }
                    }
                }
            ],
            "_bridge": {
                "username": "redacted",
                "port": 58694
            },
            "platform": "zigbee2mqtt"
        },
        {
            "name": "BraviaTVOS",
            "debug": false,
            "warn": true,
            "error": true,
            "extendedError": true,
            "tvs": [
                {
                    "active": false,
                    "polling": 10,
                    "sheduledRefresh": 12,
                    "speaker": {
                        "output": "speaker",
                        "increaseBy": 1,
                        "reduceBy": 1,
                        "accType": "lightbulb"
                    }
                },
                {
                    "active": true,
                    "name": "Sony TV",
                    "ip": "redacted",
                    "oldModel": false,
                    "port": 80,
                    "psk": "redacted",
                    "manufacturer": "Sony",
                    "model": "Bravia",
                    "serialNumber": "00000000",
                    "refreshInputs": true,
                    "wol": false,
                    "polling": 10,
                    "sheduledRefresh": 12,
                    "inputs": [
                        {
                            "name": "Nintendo Switch",
                            "identifier": "HDMI 1",
                            "source": "hdmi"
                        },
                        {
                            "name": "Xbox Series X",
                            "identifier": "HDMI 2",
                            "source": "hdmi"
                        },
                        {
                            "name": "PlayStation 5",
                            "identifier": "HDMI 3/ARC",
                            "source": "hdmi"
                        },
                        {
                            "name": "HDMI 4",
                            "identifier": "HDMI 4",
                            "source": "hdmi"
                        },
                        {
                            "name": "Video 1",
                            "identifier": "Video 1",
                            "source": "composite"
                        },
                        {
                            "name": "Video 2/Component",
                            "identifier": "Video 2/Component",
                            "source": "component"
                        }
                    ],
                    "apps": [
                        {
                            "name": "Screen Mirror",
                            "identifier": "Screen mirroring"
                        }
                    ],
                    "displayOrder": [
                        "inputs",
                        "apps"
                    ],
                    "speaker": {
                        "active": true,
                        "output": "speaker",
                        "increaseBy": 1,
                        "reduceBy": 1,
                        "accType": "lightbulb"
                    }
                }
            ],
            "_bridge": {
                "username": "redacted",
                "port": 50713
            },
            "platform": "BraviaOSPlatform"
        },
        {
            "email": "redacted@hotmail.com",
            "password": "redacted",
            "name": "Garage Door MyQ",
            "options": [
                "Disable. redacted",
                "Disable.Device. redacted"
            ],
            "refreshInterval": 12,
            "activeRefreshInterval": 3,
            "activeRefreshDuration": 302,
            "debug": true,
            "_bridge": {
                "username": "redacted,
                "port": 44864
            },
            "platform": "myQ"
        },
        {
            "name": "Nest",
            "access_token": "redacted",
            "_bridge": {
                "username": "redacted",
                " port": 47144
            },
            "platform": "Nest"
        },
        {
            "devices": [
                {
                    "name": "Xbox Series X",
                    "host": "redacted",
                    "xboxLiveId": "redacted",
                    "webApiControl": true,
                    "getInputsFromDevice": false,
                    "filterGames": false,
                    "filterApps": false,
                    "filterSystemApps": false,
                    "filterDlc": false,
                    "inputs": [
                        {
                            "name": "Netflix",
                            "reference": "4DF9E0F8.Netflix_mcm4njqhnhss8!App",
                            "oneStoreProductId": "9WZDNCRFJ3TJ",
                            "contentType": "App"
                        },
                        {
                            "name": "Prime Video",
                            "reference": "AmazonVideo.AmazonVideo_pwbj9vvecjh7j!App",
                            "oneStoreProductId": "9NBLGGH4VGD7",
                            "contentType": "App"
                        },
                        {
                            "name": "Apple TV",
                            "reference": "AppleInc.AppleTV_nzyj5cx40ttqa!App",
                            "oneStoreProductId": "9MW0ZWQFH0M2",
                            "contentType": "App"
                        },
                        {
                            "name": "Disney+",
                            "reference": "Disney.37853FC22B2CE_6rarf9sa4v8jt!App",
                            "oneStoreProductId": "9NXQXXLFST89",
                            "contentType": "App"
                        },
                        {
                            "name": "HBO Max",
                            "reference": "HomeBoxOfficeInc.HBONOW_k40401y96tbnw!App",
                            "oneStoreProductId": "9PJJ1K9DZMRS",
                            "contentType": "App"
                        },
                        {
                            "name": "Hulu",
                            "reference": "HuluLLC.HuluPlus_fphbd361v8tya!App",
                            "oneStoreProductId": "9WZDNCRFJ3L1",
                            "contentType": "App"
                        },
                        {
                            "name": "PeacockTV",
                            "reference": "PEACOCKTVLLC.PEACOCKTV_4aes6aww7z2rj!App",
                            "oneStoreProductId": "9PL67R4P9PG5",
                            "contentType": "App"
                        },
                        {
                            "name": "Movies & TV",
                            "reference": "Microsoft.ZuneVideo_8wekyb3d8bbwe!Microsoft.ZuneVideo",
                            "oneStoreProductId": "9WZDNCRFJ3P2",
                            "contentType": "App"
                        }
                    ],
                    "sensorPower": false,
                    "sensorInput": false,
                    "sensorScreenSaver": false,
                    "webApiPowerOnOff": true,
                    "xboxWebApiToken": redacted ",
                    "enableDebugMode": false,
                    "disableLogInfo": false,
                    "disableLogDeviceInfo": false,
                    "volumeControl": -1,
                    "enableMqtt": false,
                    "mqttDebug": false,
                    "mqttAuth": false
                }
            ],
            "_bridge": {
                "username": "redacted",
                "port": 59973
            },
            "platform": "XboxTv"
        },
        {
            "refreshToken": "redacted",
            "alarmOnEntryDelay": true,
            "hideDoorbellSwitch": true,
            "hideCameraSirenSwitch": true,
            "hideInHomeDoorbellSwitch": true,
            "hideAlarmSirenSwitch": true,
            "avoidSnapshotBatteryDrain": true,
            "hideDeviceIds": [
                "redacted",
                "redacted"
            ],
            "_bridge": {
                "username": "redacted",
                "port": 51132
            },
            "platform": "Ring"
        },
        {
            "pollInterval": 5000,
            "platform": "playstation",
            "_bridge": {
                "username": "redacted ",
                "port": 56159
            }
        }
    ],
    "disabledPlugins": [
        "homebridge-irobot",
        "homebridge-wemo",
        "homebridge-nest"
    ]
}

```json
Show the relevant portion of your homebridge config.json here, if needed.
Remove any sensitive information.
```

**Screenshots:**

**Environment:**

* **Homebridge Version:**  1.6.1
* **Node Version:** [18.17.1]
* **Homebridge-myQ Plugin Version**:
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** all are on 17.0.3
* **Operating System and OS Version:**  Raspbian (dont know how to look what version)

**Comment by mjoshea148** [comment]
For reference, this topic was raised (and apparently will soon be resolved) on HomeAssistant GitHub:  https://github.com/home-assistant/core/issues/101763

**Comment by mjoshea148** [comment]
Looks like it’s API/authentication related.  This is a better link to how one person apparently resolved it:

https://github.com/home-assistant/core/issues/101763#issuecomment-1756195278￼

On Oct 12, 2023, at 9:42 AM, ChandlerHooley ***@***.***> wrote:


Does the fix that they discussed in the HA issue also work as a fix/update for the home bridge side of the myq plugin? Or what would we have to do to update our side with that fix?

**Comment by BlueThunder93** [comment]
I am seeing the same issue here.  I moved away from the HA Plugin because it was becoming to unreliable and saw many suggestions to use the Homebridge MyQ plugin, so I spun HB back up and loaded the plugin - it was working for a couple of weeks until now.  Any insight to resolving this would be much appreciated.  Thanks!

**Comment by joshdavis11** [comment]
Looks like @nathangoodman submitted a pull request to address this: https://github.com/hjdhjd/myq/pull/419.

**Comment by bverenkoff** [comment]
JFYI, pasting from a Reddit post, but someone contacted myQ and they said this:

"For the safety of our MYQ Customers we did disable the use of our MYQ app with non-authorized third party apps that do not have a partnership with us to be controlling our app."

**Comment by hjdhjd** [comment]
> JFYI, pasting from a Reddit post, but someone contacted myQ and they said this:
> 
> "For the safety of our MYQ Customers we did disable the use of our MYQ app with non-authorized third party apps that do not have a partnership with us to be controlling our app."

Super helpful thanks. 😄 

Will be working through this...stay tuned.

**Comment by bverenkoff** [comment]
> > JFYI, pasting from a Reddit post, but someone contacted myQ and they said this:
> > "For the safety of our MYQ Customers we did disable the use of our MYQ app with non-authorized third party apps that do not have a partnership with us to be controlling our app."
> 
> Super helpful thanks. 😄
> 
> Will be working through this...stay tuned.

Here's the link:
https://www.reddit.com/r/homebridge/comments/175iig3/myq_login_failures/

**Comment by dreagleeyes** [comment]
In the mean time is there a way to turn off the authentication attempts short of disabling the plugin? All of the failed api attempts are  making it harder for their app to authenticate inside my network.

**Comment by hjdhjd** [comment]
Disable the child bridge, if that's how you've configured it, in the Homebridge webUI, or disable the plugin in the Homebridge webUI. That's it.

**Comment by hjdhjd** [comment]
v3.4.2 should do the trick for folks. Cheers.

**Comment by marcelgood** [comment]
Thanks @hjdhjd. v3.4.2 did the trick for now, but I'm wondering for how long if they are starting to crack down on API access. They've discontinued their Homekit bridge, so what is the future plan. Anybody know if they are adding Matter support, so we can control myQ more natively from Homekit?

**Comment by stevehoek** [comment]
Still happens for me on v3.4.2... getting 429 errors.  Do I need to disable the plugin for awhile?

**Comment by ChandlerHooley** [comment]
> Still happens for me on v3.4.2... getting 429 errors. Do I need to disable the plugin for awhile?

Mine did then when I was doing some testing. You may need to disable it for a while as it may have failed too many times. Mine did connect back up with the update, but then again it had been disabled for 12 hours. I don’t know how long you would need to but that’s just my experience.

TL;DR: Yes.

**Comment by marcelgood** [comment]
> Still happens for me on v3.4.2... getting 429 errors. Do I need to disable the plugin for awhile?

Did you restart Homebridge after updating the plugin? I don't know how long mine was hammering the API, because I only noticed that something was wrong when my garage door didn't open around noon. After finding this issue I updated to v3.4.2 and restarted Homebridge and then it immediately connected and things have been fine since then.

**Comment by hjdhjd** [comment]
You’ll need to wait 60-90 minutes if you’re seeing those errors. As I said on the Discord: myQ has significantly increased the penalty for not strictly complying with how they want you to communicate with the API. The result is a lockout for an hour or so…it’s made troubleshooting and evolving in this shifting landscape more challenging. Be patient. Wait. It will be fine in a bit. You can disable the plugin for an hour or so if it’s bugging you.

Locking this issue to avoid further necroposting.
